### DISCUSSION TITLE
Extend Theme for Input Component and extend the styling to the Right/Left Elements

### DISCUSSION THREAD

**Discussion by revathi-jayaraman** [chakra-ui/chakra-ui#5247] [Q&A]
Is there a way to extend the theme for the input component and apply the style to the Input Right/Left Elements? 

![image](https://user-images.githubusercontent.com/75802499/146088588-ddfffa84-72b6-4f62-b370-117f3cb3897c.png)

for eg if the input is Invalid apply the Icon color will be "red" 

```
<InputGroup>
    <InputLeftElement
      pointerEvents='none'
      children={<PhoneIcon />}
    />
    <Input type='tel' isInvalid placeholder='Phone number' />
  </InputGroup>

```
![image](https://user-images.githubusercontent.com/75802499/146088085-db2210ac-91b4-4709-9ac6-7d2af66981c8.png)

without passing the css for the Icon which is placed inside the InputLeftElement instead inherit the theme property from input component? 

Thanks!

**Comment by primos63** [comment]
I answered this sort of thing [here](https://github.com/chakra-ui/chakra-ui/discussions/5242)
Modified the [CodeSandbox](https://codesandbox.io/s/chakra-inputgroup-with-color-bu59g) to include an invalid state, only the left element but you can extend that to the right element.
